### ISSUE TITLE
Remove cld2 and SSE4.2 from configure.ac

### ISSUE THREAD

**Issue by anatoli26** [cyrusimap/cyrus-imapd#3102]
@elliefm,

I see that the cld2 detection code is still in configure.ac. Shouldn't it be removed based on what we've discussed recently on cyrus-devel@ (i.e. cld2 is not used anywhere and was included by mistake)?

And related, recently there was a thread ("cyrus-imap buildin: sse extention") in info-cyrus@ about SSE4.2 compilation for CRC32c algo. I've commented that maybe the detection and informational code should also be removed from configure.ac as it is not used in any way at this moment and there are no plans to make it work, so to remove the confusion and simplify the code.

I can submit a PR if you prefer (without the merge mess this time : ).

**Comment by elliefm** [comment]
> I see that the cld2 detection code is still in configure.ac. Shouldn't it be removed

Yep.  Hadn't gotten around to it yet, and then forgot.  Thanks for opening an issue! :)  I don't think it's urgent, but it would be nice to have it done eventually.  A PR would be great; otherwise I'll probably sort it out in time for next year's 3.4 release.

> SSE4.2

There's only been that one question about it in several years, so I don't think it's so confusing, nor presents such a support headache, as to be worth removing in a hurry.  Maybe if it turns out to cause actual problems, but so far it doesn't seem like it does.  (I guess it might if someone tried to cross-compile, but I'm pretty sure cross-compiling Cyrus would fail for lots of reasons, not just this thing.)

On the other hand, it's currently dead code, and if we did remove it, bringing it back from git history if we needed it again later would be relatively trivial, in which case it might as well go.  I don't really have a strong opinion either way.

Off the top of my head, I'm not sure how much work it would be to remove it cleanly, but my gut feeling is that it "figuring out how to do it" and "doing it" would not be much different.  But so far, the cost of potentially doing so has been less than the cost of answering one mailing list question about it. :)

If you wanna have a go at removing it and submit a PR, we can throw it to the rest of the Cyrus maintainers and see if anyone feels strongly about it one way or another.

If you end up doing both of these, please submit each as separate PRs, so we can merge or not-merge them independently.  Thanks!

**Comment by anatoli26** [comment]
Ellie, ok, it's clear. I'll try to prepare both PRs these days.

With respect to SSE4.2, I guess there are more people questioning/doing something about it that those asking about it on the ML. For example, the OpenBSD port builds with the option `cyrus_cv_sse42=no`. I suppose it's done to avoid problems at run-time on processors that don't have this instruction set extension and for those building it for distribution, my guess is that they all somehow disable it (or ignore it if the maintainers understand that this is actually a dead code affecting nothing).

Also, unrelated, but... there are a number of minor patches for the cyrus-imapd port on OpenBSD. May I submit them as PRs? Things like this:

```
     case "${target_os}" in
-        linux*|netbsd*|freebsd*|dragonfly*)
+        linux*|netbsd*|freebsd*|dragonfly*|openbsd*)
```

**Comment by elliefm** [comment]
Yeah, that'd be great.  If we can integrate patches that distro packages are needing to add, then that brings the distro packages closer to the pure source, and makes life easier for everyone.  We've had a few contributions along those lines in the past from Debian/Fedora maintainers, for example.

**Comment by anatoli26** [comment]
@elliefm, I just created the 2 PR to remove cld2 & SSE4.2 dependency checks.

The 1st one is very simple.

The second one was more complex: it involved a cunit test and I removed 2 entire files implementing the CRC32c functionality.

**Comment by brong** [comment]
```
squatter[482333]: IOERROR: Xapian: language indexing requires CLD2 library
```

CLD2 is not as unused as all that - it was just temporarily only in the xapian_great_reindex branch, which is coming back shortly.  I will be reverting that commit.

**Comment by anatoli26** [comment]
We need to update the compiling page then. And `configure` probably needs to check if we are building with Xapian and only then check for cld2 and for that case make it required, not optional like it was before.

**Comment by elliefm** [comment]
Was automatically closed by merging the SSE PR.  Reopening so we don't forget to update the compiling doc

**Comment by anatoli26** [comment]
@elliefm, @brong, I just created a PR (https://github.com/cyrusimap/cyrus-imapd/pull/3117) to clarify the CLD2 dep in compiling.rst. Please check if the Xapian/--enable-xapian detail is correct (I don't use it, so it's just based on your previous comment about build error).

On the other hand, IMO it's necessary to alter configure.ac to put the CLD2 check inside the Xapian-enabling code and to make it required there. With the change to remove the cld2 check reverted, it's just detected for any case and is totally optional. Ellie, I'm not that fluent with autoconf, could you please make this change?

**Comment by elliefm** [comment]
That sort of thing is usually just a mess later, especially given how complicated the xapian detection already is.  So I'm disinclined to nest the CLD2 detection into it -- a long term goal is to make configure less complicated, not more.  I think it's fine to just document that it's optional and provides feature-x, and people can read the documentation and decide whether they want that feature.

**Comment by elliefm** [comment]
I've changed my mind after discussion in the PR for the docs (#3117).  We should more strongly link this dependency check to Xapian, because behaviour of imapd.conf settings depends on its existence (and changing an imapd.conf setting should not require a rebuild).
